pyccl/halos/pk_hm.py

```diff
@@ -0,0 +1,569 @@
+    this class and overloading the :meth:`~ProfileCovar.fourier_covar`
+    method.
+    """
+    def __init__(self):
+        pass
+
+    def fourier_covar(self, prof, cosmo, k, M, a,
+                      prof_2=None, mass_def=None):
+        """ Returns the Fourier-space two-point moment between
+        two profiles:
+
+        .. math::
+           \\langle\\rho_1(k)\\rho_2(k)\\rangle.
+
+        Args:
+            prof (:class:`~pyccl.halos.profiles.HaloProfile`):
+                halo profile for which the second-order moment
+                is desired.
+            cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+            k (float or array_like): comoving wavenumber in Mpc^-1.
+            M (float or array_like): halo mass in units of M_sun.
+            a (float): scale factor.
+            prof_2 (:class:`~pyccl.halos.profiles.HaloProfile`):
+                second halo profile for which the second-order moment
+                is desired. If `None`, the assumption is that you want
+                an auto-correlation, and `prof` will be used as `prof_2`.
+            mass_def (:obj:`~pyccl.halos.massdef.MassDef`): a mass
+                definition object.
+
+        Returns:
+            float or array_like: second-order Fourier-space
+            moment. The shape of the output will be `(N_M, N_k)`
+            where `N_k` and `N_m` are the sizes of `k` and `M`
+            respectively. If `k` or `M` are scalars, the
+            corresponding dimension will be squeezed out on output.
+        """
+        if not isinstance(prof, HaloProfile):
+            raise TypeError("prof must be of type `HaloProfile`")
+        uk1 = prof.fourier(cosmo, k, M, a, mass_def=mass_def)
+
+        if prof_2 is None:
+            uk2 = uk1
+        else:
+            if not isinstance(prof_2, HaloProfile):
+                raise TypeError("prof_2 must be of type "
+                                "`HaloProfile` or `None`")
+
+            uk2 = prof_2.fourier(cosmo, k, M, a, mass_def=mass_def)
+
+        return uk1 * uk2
+
+
+class HMCalculator(object):
+    """ This class implements a set of methods that can be used to
+    compute various halo model quantities. A lot of these quantities
+    will involve integrals of the sort:
+
+    .. math::
+       \\int dM\\,n(M,a)\\,f(M,k,a),
+
+    where :math:`n(M,a)` is the halo mass function, and :math:`f` is
+    an arbitrary function of mass, scale factor and Fourier scales.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+        l10M_min (float): logarithmic mass (in units of solar mass)
+            corresponding to the lower bound of the integrals in
+            mass. Default: 8.
+        l10M_max (float): logarithmic mass (in units of solar mass)
+            corresponding to the upper bound of the integrals in
+            mass. Default: 16.
+        nl10M (int): number of samples in log(Mass) to be used in
```

Comment:
these parameter names could be more verbose and clear (e.g., log10M_min)

Reply:
Done